assignments/this.js

```diff
@@ -1,26 +1,62 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. when in the global scope, this refers to the window object.
+* 2. implicit- when a function is called with a dot, this refers to whatever is to the left of the dot.
+* 3. explicit- whenever the javascript method call or apply is used, the first argument is what this is referring to.
+* 4.  new - whenever a constructor function is used, this refers to the object created with the function.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
-
 // code example for Window Binding
+function helloWorld(planet) {
+    console.log(this); 
+    return 'hi ' + planet//? 
+};
+
+helloWorld("earth"); //?
 
-// Principle 2
 
+// Principle 2
 // code example for Implicit Binding
 
-// Principle 3
+const person = {
+    'language': 'javascript',
+    'type': 'programmer', 
+    'code': function(name) {
+        return `${name} can code in ${this.language}!`; //? 
+    }
+}
 
-// code example for New Binding
+person.code('josh'); //? 
 
-// Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// Principle 3
+// code example for New Binding
+function ninjaAttack(enemy) {
+this.scream = "HiYAAAA! "
+this.enemy = enemy;
+this.kick = function() {
+  return this.scream + this.enemy + ' was kicked!';
```

Comment:
This line is really neat & imaginative. Did you use template literals elsewhere?